internal/internal_worker.go

```diff
@@ -1122,6 +1144,27 @@ func isInterfaceNil(i interface{}) bool {
 	return i == nil || reflect.ValueOf(i).IsNil()
 }
 
+func getReadOnlyChannel(c chan struct{}) <-chan struct{} {
+	return c
+}
+
+func withWorkerStopChannel(parent context.Context, c <-chan struct{}) (context.Context, context.CancelFunc) {
+	if parent == nil {
+		parent = context.Background()
+	}
+
+	ctx := context.WithValue(parent, workerShutdownChannelContextKey, &c)
```

Comment:
Why is the shutdown channel is not part of the activityEnvironment structure?

Reply:
The activity environment is generated in the Execute method in the activity task handler. So I use context to pass the channel the task handler. Another option is to put the channel in workerExecutionParameters and then pass it to the activity environment. Any concern to use context?

Reply:
I think workerExecutionParams makes more sense. I would prefer having all of the activity related data to be part of the activityEnvironment instead of multiple context keys.

Reply:
Ok. Then I will move it to workerExecutionParameters.